tests/test_centrality.py

```diff
@@ -0,0 +1,79 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestCentralityDiGraph(unittest.TestCase):
+    def setUp(self):
+        self.graph = retworkx.PyDiGraph()
+        self.a = self.graph.add_node("A")
+        self.b = self.graph.add_node("B")
+        self.c = self.graph.add_node("C")
+        self.d = self.graph.add_node("D")
+        edge_list = [
+            (self.a, self.b, 1),
+            (self.b, self.c, 1),
+            (self.c, self.d, 1),
+        ]
+        self.graph.add_edges_from(edge_list)
+
+    def test_betweenness_centrality(self):
+        betweenness = retworkx.digraph_betweenness_centrality(self.graph)
+        expected = [0.0, 0.3333333333333333, 0.3333333333333333, 0.0]
+        self.assertEqual(expected, betweenness[0])
+
+    def test_betweenness_centrality_endpoints(self):
+        betweenness = retworkx.digraph_betweenness_centrality(
+            self.graph, endpoints=True)
+        expected = [0.25, 0.41666666666666663, 0.41666666666666663, 0.25]
+        self.assertEqual(expected, betweenness[0])
+
+    def test_betweenness_centrality_unnormalized(self):
+        betweenness = retworkx.digraph_betweenness_centrality(
+            self.graph, endpoints=False, normalized=False)
+        expected = [0.0, 2.0, 2.0, 0.0]
+        self.assertEqual(expected, betweenness[0])
+
```

Comment:
It probably would be good to have a test case here for holes in the node indices. https://github.com/Qiskit/retworkx/pull/229/files#diff-abab60754038a6bbd6ef6b34128658a3bedbdbccc6743e87c5f8a2d6660f5614R113 is an example where I just did this in an open PR.

Reply:
For the same graph you build in your test, my current implementation results in `panicked at 'index out of bounds`. I definitely need to add the test and then fix the implementation.

Reply:
The link above for the test for holes is broken (don't know if it is temporary github disruption or not). In any case, I added another test to check that the algorithm works with a hole in the index set.

Reply:
Ah yeah that test file moved around since things reorganized https://github.com/Qiskit/retworkx/blob/main/tests/graph/test_max_weight_matching.py#L242-L279 but yeah you've got the coverage below in `TestCentralityDiGraphDeletedNode`